masonry/src/widgets/progress_bar.rs

```diff
@@ -33,11 +33,11 @@ pub struct ProgressBar {
 impl ProgressBar {
     /// Create a new `ProgressBar`.
     ///
-    /// `progress` is a number between 0 and 1 inclusive. If it is `NaN`, then an
-    /// indefinite progress bar will be shown.
-    /// Otherwise, the input will be clamped to [0, 1].
-    pub fn new(mut progress: Option<f64>) -> Self {
-        clamp_progress(&mut progress);
+    /// The progress value will be clamped to [0, 1].
+    ///
+    /// A `None` value (or NaN) will show an indeterminate progress bar.
```

Comment:
I'm not a huge fan of NaN being meaningful API-wise; but that isn't for this PR to address.